org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/PlexusContainerManager.java

```diff
@@ -119,23 +150,35 @@ public class PlexusContainerManager {
   @Deactivate
   void dispose() {
     synchronized(containerMap) {
-      containerMap.values().forEach(PlexusContainer::dispose);
+      containerMap.values().forEach(PlexusContainerManager::disposeContainer);
       containerMap.clear();
       if(nonRootedContainer != null) {
-        nonRootedContainer.dispose();
+        disposeContainer(nonRootedContainer);
         nonRootedContainer = null;
       }
     }
   }
 
+  private static void disposeContainer(PlexusContainerWithExtensions container) {
+    //TODO: dispose  container.container().getContainerRealm() as well?
```

Comment:
@laeubi do you think that would be useful reasonable? Or can this cause problems?
I'm not very familiar with the ClassRealm lifecycle, but since it is created by M2E below, I wonder if it should be cleaned up by M2E as well, when the container is disposed.